syntax = "proto2";
enum BrakeNodeStates {
    bnsBooting   = 0;
    bnsStandby   = 1;
    bnsArming    = 2;
    bnsArmed     = 3;
    bnsFlight    = 4;
    bnsBraking   = 5;
    bnsVenting   = 6;
    bnsRetrieval = 7;
    bnsError     = 8;
}

enum PodStates {
    psBooting       = 0;
    psStandby       = 1;
    psArming        = 2;
    psArmed         = 3;
    psPreFlight     = 4;
    psAcceleration  = 5;
    psCoasting      = 6;
    psBraking       = 7;
    psDisarming     = 8;
    psRetrieval     = 9;
    psEmergency     = 10;
}

enum MotorStates {
    msOff       = 0;
    msCharging  = 1;
    msIdle      = 2;
    msEngaged   = 3;
}

enum ControlsInterfaceStates {
    ciArm = 0;
    ciFlight = 1;
    ciEmergencyStop = 2;
}


enum InverterStates {
    iConnected = 0;

}

enum BatteryManagementStates {
    bmsNominal = 0;
}

enum RearNodeStates {
    rnBooting = 0;
    rnNominal = 1;
}

enum LvdcNodeStates {
    lvdcBooting = 0;
}